.github/workflows/main.yml

```diff
@@ -28,3 +28,8 @@ jobs:
       SCRIBE_GRAPHQL_ACCESS_TOKEN: ${{ secrets.SCRIBE_GRAPHQL_ACCESS_TOKEN }}
     steps:
       - run: bash ./.github/scripts/run-docker.sh /runner/.github/scripts/upload-scribe.sh
+  upload-github:
+    needs: run-benchmark
+    runs-on: [self-hosted, bm-ec2]
```

Comment:
i'm wondering if this approach of separate steps that run on the same label is robust?

if we add more jobs that use the runner, or if we add more runners for the job to share, will it be possible to hit either of these scenarios?
 - that 'run-benchmark' runs on worker0, then upload-scribe runs on worker1
 - a race condition where upload-scribe picks up a file generated by another job and stored on the worker?

Reply:
Good point. Unfortunately I don't find a way to assign runner on the workflow level. Therefore, if more runners are added to "bm-ec2" label it is possible for scenario 1) to happen.
However, for 2), since upload-scribe.sh uploads the file by GITHUB_RUN_ID, it is impossible to pick up json from another job.

Reply:
ok, it'd be good to confirm if the dependency gaurantees co-scheduling or not.  

if not, it could be that you just need to combine all the steps as part of one script so they get scheduled atomically for sure?

Reply:
I asked a question here: https://github.community/t/does-gh-actions-co-schedule-dependent-jobs-on-the-same-runner/146913. Will also talk with github people when I have a chance.